tests/validation/validate_ball_and_stick.cpp

```diff
@@ -21,66 +21,75 @@
 
 using namespace nest::mc;
 
-// TODO: further consolidate common code
+struct sampler_info {
+    const char* label;
+    cell_member_type probe;
+    simple_sampler sampler;
+};
+
+template <
+    typename lowered_cell,
+    typename SamplerInfoSeq
+>
+void run_ncomp_convergence_test(
+    const char* model_name,
+    const char* ref_data_path,
+    const cell& c,
+    SamplerInfoSeq& samplers,
+    float t_end=100.f,
+    float dt=0.001)
+{
+    using nlohmann::json;
+
+    SCOPED_TRACE(model_name);
 
-TEST(ball_and_stick, neuron_ref) {
-    // compare voltages against reference data produced from
-    // nrn/ball_and_stick.py
-
-    using namespace nlohmann;
-
-    using lowered_cell = fvm::fvm_multicell<double, cell_local_size_type>;
     auto& V = g_trace_io;
-
     bool verbose = V.verbose();
     int max_ncomp = V.max_ncomp();
 
-    // load validation data
-    auto ref_data = V.load_traces("neuron_ball_and_stick.json");
-    bool run_validation =
-        ref_data.count("soma.mid") &&
-        ref_data.count("dend.mid") &&
-        ref_data.count("dend.end");
+    auto keys = util::transform_view(samplers,
+        [](const sampler_info& se) { return se.label; });
 
-    EXPECT_TRUE(run_validation);
+    bool run_validation = false;
+    std::map<std::string, trace_data> ref_data;
+    try {
+        ref_data = V.load_traces(ref_data_path);
 
-    // generate test data
-    cell c = make_cell_ball_and_stick();
-    add_common_voltage_probes(c);
+        run_validation = std::all_of(keys.begin(), keys.end(),
+            [&](const char* key) { return ref_data.count(key)>0; });
 
-    float sample_dt = .025;
-    std::pair<const char *, simple_sampler> samplers[] = {
-        {"soma.mid", simple_sampler(sample_dt)},
-        {"dend.mid", simple_sampler(sample_dt)},
-        {"dend.end", simple_sampler(sample_dt)}
-    };
+        EXPECT_TRUE(run_validation);
+    }
+    catch (std::runtime_error&) {
```

Comment:
After gtest has been informed of the failure, the simulation is still run (i.e. we don't just return). Is the idea that we still want to collect the results from NestMC?